### ISSUE TITLE
Reconsider and evaluate top-level imports

### ISSUE THREAD

**Issue by DigiDuncan** [pythonarcade/arcade#2402]
Leaving this here so we don't forget it. We need to take a quick lookover on what should be a top-level import, both for clarity, and to possibly reduce build size when `arcade` is imported.

**Comment by pushfoo** [comment]
I'm marking the evaluation of top-level imports as 3.0 mandatory since it's a commitment we have to keep to. Build size / packaging should be a separate issue.

**Comment by einarf** [comment]
I **suspect** input and media is adding a ~5 second startup time for me on my new laptop. Possibly also text. We have talked about this many times but I'm not sure if there is a clean solution. The rest of the imports I don't think cause any issues.

**Comment by cspotcode** [comment]
Some python libraries use tricks so that imports load on-demand, completely seamless from the consumer perspective.  So if you never access a heavyweight component, it never imports.

**Comment by pushfoo** [comment]
> Possibly also text. We have talked about this many times but I'm not sure if there is a clean solution.

I thought we stopped, but do we still load the Kenny fonts by default?

> Some python libraries use tricks so that imports load on-demand, completely seamless from the consumer perspective. So if you never access a heavyweight component, it never imports.

Yep, that's pyglet. They also do some interesting things like having `Window` change per-platform.

**Comment by einarf** [comment]
I put this past 3.0 for now. It's high pri for later.

**Comment by DigiDuncan** [comment]
I think this should be in 3.0 mandatory, personally. Given that we're waiting on Pyglet 2.1 anyway, taking a day to do this while we can seems important. It _can't_ reasonably be 3.0.x, since it'd be a breaking API change, so we'd have to wait until 3.1, and I'd rather not.

**Comment by einarf** [comment]
Import time on windows with my current project(s)

* Initial run when OS have flushed caches: 5.82s
* Next runs: 1.75s

We can probably get this down below a second with lazy imports but it will still be a problem for projects with text and controller input.
